### ISSUE TITLE
Option to use a fixed .zip to restore instead of folder location for Restore

### ISSUE THREAD

**Issue by unlocKing** [robweber/xbmcbackup#101]
Is it possible to add an option to choose a fixed .zip name rather than a folder to choose from?

So I can upload 1 file with all my changes that can be used to for restoring.

Cheers!

**Comment by robweber** [comment]
You could script it. The README has more information but if you know the zip file name ahead of time you could write a script, or issue like a CURL command, to the JSON-RPC to kick off a restore. Something like this: 

```
{ "jsonrpc": "2.0", "method": "Addons.ExecuteAddon","params":{"addonid":"script.xbmcbackup","params":{"mode":"restore","archive":"archive.zip"}}, "id": 1 }
```

Is that sort of what you mean? Bypass the Restore select option and just pick this specific folder name each time?

**Comment by unlocKing** [comment]
"Bypass the Restore select option and just pick this specific folder name each time?"

Specific .zip each time

**Comment by unlocKing** [comment]
RunScript(script.xbmcbackup,mode=backup)

So use that with the actual filename (.zip)

Something like:

RunScript(script.xbmcbackup,mode=backup,201702100000.zip)

Would that be possible?

**Comment by robweber** [comment]
Yes, you can pass in the name of the archive file so that it gets used as part of the restore, bypassing the selection dialog.

**Comment by unlocKing** [comment]
Sorry, I meant how can I use that so it pulls from my server?

In the Browse Remote Path, I currently have it set to:

W:\Kodi\Backup

What I'm trying to achieve is so it points directly:

W:\Kodi\Backup\201702100000.zip

Can some code be added or modified for it to just choose that filename?

Thanks.

**Comment by robweber** [comment]
This is kind of old so I'm going to close it out. I think all the info you need is in the comments though. The path is just the path to the "root". If you script it the way I mention you can have the restore point always be the same. If you really need to modify the code (ie you don't want to write a script or do a JSON call) take a look at https://github.com/robweber/xbmcbackup/blob/master/default.py#L47

Basically cut out lines 47-70 and hardcode the name into line 71. Of course this will get changed whenever you update the addon unless you do some other stuff to make a new version (change addon.xml).
